pkg/domain/domain.go

```diff
@@ -2751,15 +2770,37 @@ const (
 	tiflashComputeNodeKey = "/tiflash/new_tiflash_compute_nodes"
 )
 
+func encodeUserList(userList []string) string {
+	// use base64 encoding for the user and use ',' to separate them.
+	// Because the user name itself may contains special char like ','
+	encoded := make([]string, 0, len(userList))
+	for _, user := range userList {
+		encoded = append(encoded, base64.StdEncoding.EncodeToString([]byte(user)))
+	}
+	return strings.Join(encoded, ",")
+}
+
+func decodeUserList(userList []string, val string) []string {
+	users := strings.Split(val, ",")
+	for _, userRaw := range users {
+		user, err := base64.StdEncoding.DecodeString(userRaw)
+		if err == nil {
+			userList = append(userList, string(user))
+		}
+	}
+	return userList
+}
```

Comment:
please use a json as the value of the etcd key, easy to en/decode and extend

also what's the size of the list, etcd is not fit to store large data, max 1.5m by default